scripts/scr_ui_display_weapons/scr_ui_display_weapons.gml

```diff
@@ -8,372 +8,713 @@ function scr_ui_display_weapons(argument0, argument1, argument2) {
+			ui_weapon[rl]=spr_weapon_powfist3;
+			clear=true;
+		}
+	    if (termi_tartaros=2) and (argument0=2){
+			ui_xmod[rl]=-1;
+			ui_ymod[rl]=10;
+			fix_right=8;
+			ui_weapon[rl]=spr_weapon_powfist3;
+			clear=true;
+		}
 	}
 	if (display_type="lightning_claw"){
-	    if (argument1=0) and (argument0=1){ui_xmod[rl]+=11;}
-	    if (argument1=0) and (argument0=2){ui_xmod[rl]-=8;}
-
-	    if (argument1>0){ui_arm[rl]=false;ui_above[rl]=true;}
-
-	    if (argument1=1) and (argument0=1){ui_xmod[rl]=-3;ui_ymod[rl]=10;fix_left=8.1;ui_weapon[rl]=spr_weapon_lightning2;clear=true;}
-	    if (argument1=1) and (argument0=2){ui_xmod[rl]=2;ui_ymod[rl]=10;fix_right=8.1;ui_weapon[rl]=spr_weapon_lightning2;clear=true;}
-    
-	    if (argument1=2) and (argument0=1){ui_xmod[rl]=0;ui_ymod[rl]=10;fix_left=8.1;ui_weapon[rl]=spr_weapon_lightning2;clear=true;}
-	    if (argument1=2) and (argument0=2){ui_xmod[rl]=-1;ui_ymod[rl]=10;fix_right=8.1;ui_weapon[rl]=spr_weapon_lightning2;clear=true;}
-	}
-
-
-
-
-	if (string_count("Storm Shield",wee)>0){ui_weapon[rl]=spr_weapon_storm;
-	    ui_arm[rl]=false;ui_above[rl]=true;ui_spec[rl]=false;
-	}
-		if (string_count("Boarding Shield",wee)>0){ui_weapon[rl]=spr_weapon_boarding;
-	    ui_arm[rl]=false;ui_above[rl]=true;ui_spec[rl]=false;
+	    if (termi_tartaros=0) and (argument0=1){ui_xmod[rl]+=11;}
+	    if (termi_tartaros=0) and (argument0=2){ui_xmod[rl]-=8;}
+	    if (termi_tartaros>0){
+			ui_arm[rl]=false;
+			ui_above[rl]=true;
+		}
+	    if (termi_tartaros=1) and (argument0=1){
+			ui_xmod[rl]=-3;
+			ui_ymod[rl]=10;
+			fix_left=8.1;
+			ui_weapon[rl]=spr_weapon_lightning2;
+			clear=true;
+		}
+	    if (termi_tartaros=1) and (argument0=2){
+			ui_xmod[rl]=2;
+			ui_ymod[rl]=10;
+			fix_right=8.1;
+			ui_weapon[rl]=spr_weapon_lightning2;
+			clear=true;
+		}
+	    if (termi_tartaros=2) and (argument0=1){
+			ui_xmod[rl]=0;
+			ui_ymod[rl]=10;
+			fix_left=8.1;
+			ui_weapon[rl]=spr_weapon_lightning2;
+			clear=true;
+		}
+	    if (termi_tartaros=2) and (argument0=2){
+			ui_xmod[rl]=-1;
+			ui_ymod[rl]=10;
+			fix_right=8.1;
+			ui_weapon[rl]=spr_weapon_lightning2;
+			clear=true;
+		}
+	}
+	if (string_count("Storm Shield",equiped_weapon)>0){
+		ui_weapon[rl]=spr_weapon_storm;
+	    ui_arm[rl]=false;
+		ui_above[rl]=true;
+		ui_spec[rl]=false;
+	}
+	if (string_count("Boarding Shield",equiped_weapon)>0){
+		ui_weapon[rl]=spr_weapon_boarding;
+	    ui_arm[rl]=false;
+		ui_above[rl]=true;
+		ui_spec[rl]=false;
 	}
-
 	// Flip for offhand
-	if (argument0=2) /*and (argument1=0)*/ and (ui_xmod[rl]<0) and (display_type!="power_fist") and (display_type!="lightning_claw") then ui_xmod[rl]=ui_xmod[rl]*-1;
-
-
-
-
+	if (argument0=2) /*and (termi_tartaros=0)*/ and (ui_xmod[rl]<0) and (display_type!="power_fist") and (display_type!="lightning_claw") then ui_xmod[rl]=ui_xmod[rl]*-1;
 
+	// TODO Why is there an exit here? Do you know why? Is the code bellow a WIP? is the exit supposed to be inside something else?
 	exit;
```

Comment:
why is there an exit; here?

Reply:
not sure, need to wait for @ZAB909 input